crates/interpreter/src/side_table.rs

```diff
@@ -12,14 +12,74 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use core::slice;
+
+use bytemuck::cast;
+
 use crate::bit_field::*;
 use crate::error::*;
+use crate::parser;
+use crate::toctou::Use;
+
+pub type Parser<'m> = parser::Parser<'m, Use>;
+
+#[allow(dead_code)]
+pub struct SideTable<'m> {
+    indices: &'m [u16], // including 0 and the length of metadata_array
+    metadata: &'m Metadata<'m>,
+}
+#[allow(dead_code)]
+impl<'m> SideTable<'m> {
+    fn metadata(&self, func_idx: usize) -> Metadata<'m> {
+        Metadata(
+            &self.metadata.0
+                [self.indices[func_idx] as usize .. self.indices[func_idx + 1] as usize],
+        )
+    }
+}
+
+#[allow(dead_code)]
+struct Metadata<'m>(&'m [u8]);
+
+#[allow(dead_code)]
+impl<'m> Metadata<'m> {
+    pub fn type_idx(&self) -> usize {
+        let bytes = &self.0[.. size_of::<usize>()];
+        usize::from_ne_bytes(bytes.try_into().unwrap())
+    }
+
+    pub fn parser(&self) -> Parser<'m> {
+        unsafe {
+            Parser::new(&self.0[self.parser_size_start() + size_of::<u32>() .. self.parser_end()])
```

Comment:
This is wrong. The parser data is not in the side-table but in the module. The signature should be `fn parser(self, module: &'m [u8]) -> Parser<'m>` or probably better `fn parser(self, code: &'m [u8]) -> Parser<'m>` where `code` is the code section of the module, so we can save some range.

Also, let's just use constants for u16 offsets in the metadata:
- 0 for type idx
- 1 for parser start
- 3 for parser end
- 5 for branch table